test/parallel/test-http-allow-content-length-304.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that the http-parser doesn't expect a body when
+// a 304 Not Modified response has a non-zero Content-Length header
+
+const http = require('http');
+
+const server = http.createServer(common.mustCall((req, res) => {
+  res.setHeader('Content-Length', 11);
+  res.statusCode = 304;
+  res.end(null);
+}));
+
+server.listen(0, () => {
+  const request = http.request({
+    port: server.address().port,
+  });
+
+  request.on('response', common.mustCall((response) => {
```

Comment:
Should also include something like this in the response handler:

```js
assert.strictEqual(response.headers['content-length'], '11');
```

and maybe verify the status code too?

Reply:
Done, thanks for the review :+1: